python/cudf/cudf/core/indexing.py

```diff
@@ -164,10 +165,18 @@ def __setitem__(self, key, value):
         self._sr.iloc[key] = value
 
     def _loc_to_iloc(self, arg):
-        from cudf.core.column import column
-        from cudf.core.series import Series
-
         if is_scalar(arg):
+            if not is_numerical_dtype(self._sr.index.dtype):
+                if pd.api.types.is_integer(arg):
+                    found_index = arg
+                    return found_index
+
+                elif isinstance(
+                    arg, cudf.Scalar
+                ) and pd.api.types.is_integer_dtype(arg.dtype):
+                    found_index = arg.value
+                    return found_index
```

Comment:
I think @shwina recently wrote a utility that more gracefully handles this situation that should be used here instead

Reply:
The [PR](https://github.com/rapidsai/cudf/pull/7078) for `cudf.utils.dtypes.is_integer_dtype` hasn't been merged yet (will fix when available)